### PR TITLE
session: make unit test TestErrorRollback more stable

### PR DESCRIPTION
**Pull request by tiancaiamao**
Sometime this test fail, because retry count reach the limit (default is 10)
Make it more stable.

@coocood @AndreMouche

### CODE REVIEW

**Comment by shenli** [comment]
LGTM
But I think the latest changes would make transactions vulnerable to conflicts.

File: session/session_test.go [hunk 1]
```diff
@@ -164,6 +164,7 @@ func (s *testSessionSuite) TestErrorRollback(c *C) {
 		go func() {
 			defer wg.Done()
 			localTk := testkit.NewTestKitWithInit(c, s.store)
+			localTk.MustExec("set @@session.tidb_retry_limit = 9999999")
```
Comment (ngaut) [inline session/session_test.go]: This is a bad idea, why we need to retry so many times?

Comment (ngaut) [reply session/session_test.go]: I guess it also confuses other people, which retry count is expected? Can it work if we set it to a smaller value?

**Comment by coocood** [comment]
@tiancaiamao I think 100 times is large enough

**Comment by coocood** [comment]
@ngaut PTAL

**Pull request merged**